gradle/publish.gradle

```diff
@@ -0,0 +1,122 @@
+apply plugin: 'maven-publish'
+apply plugin: 'com.jfrog.bintray'
+apply plugin: 'com.jfrog.artifactory'
+apply from: file("${rootDir}/gradle/artifact-settings.gradle")
+
+afterEvaluate {
+    publishing {
+        publications {
+            release(MavenPublication) {
+                from components.release
+                groupId project.ext.mapboxArtifactGroupId
+                artifactId project.ext.mapboxArtifactId
+                version project.ext.versionName
+
+                artifact(androidSourcesJar)
+                artifact(androidJavadocsJar)
+
+                pom.withXml {
+                    final mainNode = asNode()
+                    mainNode.appendNode('name', project.ext.mapboxArtifactTitle)
+                    mainNode.appendNode('description', project.ext.mapboxArtifactDescription)
+                    mainNode.appendNode('url', project.ext.mapboxArtifactUrl)
+
+                    final licenseNode = mainNode.appendNode('licenses').appendNode('license')
+                    licenseNode.appendNode('name', project.ext.mapboxArtifactLicenseName)
+                    licenseNode.appendNode('url', project.ext.mapboxArtifactLicenseUrl)
+                    licenseNode.appendNode('distribution', "repo")
+
+                    final developerNode = mainNode.appendNode('developers').appendNode('developer')
+                    developerNode.appendNode('id', project.ext.mapboxDeveloperId)
+                    developerNode.appendNode('name', project.ext.mapboxDeveloperName)
+
+                    final scmNode = mainNode.appendNode("scm")
+                    scmNode.appendNode("connection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("developerConnection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("url", project.ext.mapboxArtifactUrl)
+                }
+            }
+            debug(MavenPublication) {
+                from components.debug
+                groupId project.ext.mapboxArtifactGroupId
+                artifactId project.ext.mapboxArtifactId
+                version project.ext.versionName
+
+                artifact(androidSourcesJar)
+                artifact(androidJavadocsJar)
+            }
+        }
+    }
+}
+
+bintray {
+    user = mapboxBintrayUser
```

Comment:
user and key mustn't by null or empty so may be add check for it and trows obvious exception.

Reply:
better place is `artifact-settings.gradle` file, wdyt?

Reply:
This variable is coming from `artifact-settings.gradle`